### ISSUE TITLE
golangci: reenable go-critic when it works with generics

### ISSUE THREAD

**Issue by mmatczuk** [scylladb/scylla-manager#3202]
The [issue]( https://github.com/go-critic/go-critic/issues/1224) in the linter makes it impossible to run `make .check-lint`. When the issue is gone go-critic needs to be reenabled.

**Comment by mmatczuk** [comment]
```
ERRO [runner] Panic: gocritic: package "scheduler" (isInitialPkg: true, needAnalyzeSource: true): assertion failed: goroutine 49423 [running]:
runtime/debug.Stack()
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/runtime/debug/stack.go:24 +0x68
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1()
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/goanalysis/runner_action.go:101 +0x104
panic({0x10529dc60, 0x10544a750})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/runtime/panic.go:838 +0x204
go/types.assert(...)
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/types/errors.go:21
go/types.(*StdSizes).Alignof(0x140001b52d0, {0x105452500?, 0x1402ea0c750})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/types/sizes.go:72 +0x278
go/types.(*StdSizes).Offsetsof(0x105452398?, {0x1402eb7b890, 0x5, 0x0?})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/types/sizes.go:101 +0xcc
go/types.(*StdSizes).Sizeof(0x140001b52d0, {0x105452398?, 0x1402ea57650})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/types/sizes.go:154 +0x1a8
github.com/go-critic/go-critic/checkers.(*rangeValCopyChecker).VisitStmt(0x140165da880, {0x1054558c8?, 0x1402e0db270?})
        /Users/michal/go/pkg/mod/github.com/go-critic/go-critic@v0.6.3/checkers/rangeValCopy_checker.go:68 +0xb0
github.com/go-critic/go-critic/checkers/internal/astwalk.(*stmtWalker).WalkFile.func1({0x105451fb0?, 0x1402e0db270?})
        /Users/michal/go/pkg/mod/github.com/go-critic/go-critic@v0.6.3/checkers/internal/astwalk/stmt_walker.go:23 +0x64
go/ast.inspector.Visit(0x140261bc910, {0x105451fb0?, 0x1402e0db270?})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/ast/walk.go:387 +0x3c
go/ast.Walk({0x10544f4d8?, 0x140261bc910?}, {0x105451fb0?, 0x1402e0db270?})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/ast/walk.go:52 +0x50
go/ast.walkStmtList({0x10544f4d8, 0x140261bc910}, {0x1402e5c06a0?, 0x2, 0x1400d30b778?})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/ast/walk.go:32 +0x90
go/ast.Walk({0x10544f4d8?, 0x140261bc910?}, {0x105451920?, 0x1402e5bd6b0?})
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/ast/walk.go:235 +0xea4
go/ast.Inspect(...)
        /opt/homebrew/Cellar/go/1.18.3/libexec/src/go/ast/walk.go:399
github.com/go-critic/go-critic/checkers/internal/astwalk.(*stmtWalker).WalkFile(0x1401659d1b0?, 0x1402e097100?)
        /Users/michal/go/pkg/mod/github.com/go-critic/go-critic@v0.6.3/checkers/internal/astwalk/stmt_walker.go:21 +0x144
github.com/go-critic/go-critic/framework/linter.(*Checker).Check(...)
        /Users/michal/go/pkg/mod/github.com/go-critic/go-critic@v0.6.3/framework/linter/linter.go:129
github.com/golangci/golangci-lint/pkg/golinters.runGocriticOnFile(0x140325796e0, 0x140012bf232?, {0x140253e6c00, 0x61, 0x0?})
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/gocritic.go:181 +0xa4
github.com/golangci/golangci-lint/pkg/golinters.runGocriticOnPackage(0x140325796e0, {0x140253e6c00, 0x61, 0x80}, {0x1402e0ea360, 0x4, 0x14003b2d3f8?})
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/gocritic.go:169 +0x15c
github.com/golangci/golangci-lint/pkg/golinters.NewGocritic.func1.1(0x14032c79790)
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/gocritic.go:51 +0x1d8
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0x1400465c6a0)
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/goanalysis/runner_action.go:187 +0x96c
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func2()
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/goanalysis/runner_action.go:105 +0x24
github.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0x140015e37c0, {0x104fa3740, 0x8}, 0x14003b2d730)
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/timeutils/stopwatch.go:111 +0x48
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0x3?)
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/goanalysis/runner_action.go:104 +0x78
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func2(0x1400465c6a0)
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/goanalysis/runner_loadingpackage.go:80 +0xb4
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze
        /Users/michal/go/pkg/mod/github.com/golangci/golangci-lint@v1.46.2/pkg/golinters/goanalysis/runner_loadingpackage.go:75 +0x16c 
```

**Comment by mmatczuk** [comment]
@Michal-Leszczynski let's disable go-critic for now.

**Comment by Michal-Leszczynski** [comment]
Closed in 389b4082 (#3500).
